### DISCUSSION TITLE
Powershell remoting (enter-pssession)?

### DISCUSSION THREAD

**Discussion by evolite** [gravitational/teleport#32734] [Q&A] [answered]
Hi,

I'm currently investigating using Teleport Enterprise for a number of environments.

Is it possible / is it on the roadmap atm to provide support for commandline access to windows hosts?
Is there any alternatives to this atm? I saw someone mention open-ssh on windows, but preferrably we would be able to remote control windows machines using native powershell commands just like with ssh on Linux.

Thanks

**Comment by webvictim** [comment, marked as answer]
It isn't currently possible to do this even when using OpenSSH on Windows, unfortunately. The reason is that the Windows builds of the OpenSSH Server (`sshd`) have support for certificate-based authentication deliberately removed for some reason, meaning Teleport can't connect to an SSH server on a Windows host even if configured correctly. This appears to be something of a bizarre decision by Microsoft (in their infinite wisdom)

We have an open feature request for supporting WinRM via Teleport - please feel free to add a +1 and comment if it's something you'd like to see: https://github.com/gravitational/teleport/issues/9567

In the meantime, one thing you could do would likely be to set up WinRM independently and then use [Teleport's TCP application access](https://goteleport.com/docs/application-access/guides/tcp/) to get authenticated tunnels to the WinRM port on the hosts you need. It's crude, but it could potentially handle connectivity for you if your servers are in distributed environments.
